build/patch-container-image.sh

```diff
@@ -27,6 +27,28 @@ function patch_container_image() {
     fi
 }
 
+function patch_oauth_proxy_image() {
+    local container=$1
+    local file=$2
+    if [ -f "$file" ]; then
+        if grep -q 'if contains "/" .Values.global.oauthproxy.image' "${file}"; then return 0; fi
+        sed_wrap -i -re '/^ *containers: *$/,/^ *volumes:/ {
+            /- name: *'"${container}"' *$/,/imagePullPolicy:/ {
+                /name:|imagePullPolicy:/!d
+                /imagePullPolicy:/s+^(( *)imagePullPolicy:.*)$+\{\{- if contains "/" .Values.global.oauthproxy.image \}\}\
+\2image: \{\{ .Values.global.oauthproxy.image \}\}\
+\{\{- else \}\}\
+\2image: {{ .Values.global.oauthproxy.hub }}/{{ .Values.global.oauthproxy.image }}:{{ .Values.global.oauthproxy.tag }}\
+\{\{- end \}\}\
+\1+
+            }
+        }' "${file}"
+    else
+        echo "ERROR: file does not exist: ${file}"
+        return 1
+    fi
+}
```

Comment:
IIRC we're injecting oauth-proxy in another sed call, wouldn't it be easier to change it there?

Reply:
I updated it in those scripts.  This still needs to hang around to patch the old 1.0 charts, unfortunately.

Reply:
gotcha, missed that. Thanks!